odoo/addons/base/static/tests/base_settings_tests.js

```diff
@@ -157,6 +157,44 @@ QUnit.module('base_settings_tests', {
         form.destroy();
     });
 
+    QUnit.test('hide / show setting tips properly', async function (assert) {
+        assert.expect(3);
+
+        const form = await createView({
+            View: BaseSettingsView,
+            model: 'res.config.settings',
+            data: this.data,
+            arch: `
+                <form string="Settings" class="oe_form_configuration o_base_settings">
+                    <div class="o_panel">
+                        <div class="setting_search">
+                            <input type="text" class="searchInput" placeholder="Search..." />
+                        </div>
+                    </div>
+                    <div class="o_setting_container">
+                        <div class="settings_tab" />
+                        <div class="settings">
+                            <div class="notFound o_hidden">No Record Found</div>
+                            <div class="app_settings_block" string="CRM" data-key="crm">
+                                <h2>Setting Header</h2>
+                                <h3 class="o_setting_tip">CRM settings will appear below</h3>
+                            </div>
+                        </div>
+                    </div>
+                </form>`
+        });
+
+        assert.containsOnce(form, '.o_setting_tip:not(.o_hidden)', "Tip should not be hidden initially");
+
+        await testUtils.fields.editAndTrigger(form.$('.searchInput'), 'trea', 'keyup');
+        assert.containsOnce(form, '.o_setting_tip.o_hidden', "Tip should be hidden when user searches in settings");
```

Comment:
maybe in your search can you include the text of this tips ?

Reply:
We don't search the strings/headings and if try to search it we don't get any results. We are only searching the fields and whichever result field we get is under the header of its module name. And searching of field is already covered in another test case.
I am not clear exactly what you want here? Can you please explain to me if you want anything else?
Thanks in advance!

Reply:
I mean search a part of the tips text

Reply:
We cannot search for that. We can only search any field name/label.

Reply:
is your test failing without your change ?

Reply:
This is our test case failing without any changes in code.

![Screenshot from 2022-05-05 11-57-22](https://user-images.githubusercontent.com/72728567/166872434-4012e2f6-2b87-46ad-a1b8-3e260681f9fc.png)

Reply:
ah ok, for the initial search, but it still don't understand why you can't search the string you want... "trea" already match nothing

Reply:
I meant to say that the existing searching mechanism only searches on field labels, not ANY of the helper strings, Headers (or even buttons below the fields). See this video (from the master branch): https://youtu.be/5pJxv59qeMM 

Our task is about hiding the technical tip during the search, not about changing the search mechanism to match the helper strings / headers / technical tips / buttons.

And, searching based on the field is already covered in another test case. Do you want me to do the same in our test case also (that search string should match the field etc)?

Reply:
Yes I know, you don't change the search engine

But before, if your search doesn't match your tips, the tips won't be visible

So the end of your test doesn't fail without your change

Reply:
I just want to make the test more robust, and to ensure that the tips is not visible even if the search match the text in it

Reply:
Done!